Assets/Scripts/Characters/Human/TriggerColliderWeapon.cs

```diff
@@ -9,360 +9,238 @@ public class TriggerColliderWeapon : MonoBehaviour
 {
     public Equipment Equipment { get; set; }
 
-    public bool active_me;
-    public GameObject currentCamera;
+    public bool IsActive;
+    public IN_GAME_MAIN_CAMERA currentCamera;
     public ArrayList currentHits = new ArrayList();
     public ArrayList currentHitsII = new ArrayList();
     public AudioSource meatDie;
+    public Hero hero;
     public int myTeam = 1;
     public float scoreMulti = 1f;
+    public Rigidbody body;
 
-    private bool checkIfBehind(GameObject titan)
+    private FengGameManagerMKII manager;
+
+    private bool IsBehind(GameObject titan)
     {
         if (titan.transform.Find("Amarture/Core/Controller_Body/hip/spine/chest/neck/head") != null)
         {
             Transform transform = titan.transform.Find("Amarture/Core/Controller_Body/hip/spine/chest/neck/head");
-            Vector3 to = base.transform.position - transform.transform.position;
+            Vector3 to = transform.position - transform.transform.position;
             return (Vector3.Angle(-transform.transform.forward, to) < 70f);
         }
         else if (titan.transform.Find("BodyPivot/HeadPos") != null)// dummy titan
         {
             Transform transform = titan.transform.Find("BodyPivot/HeadPos");
-            Vector3 to = base.transform.position - transform.transform.position;
+            Vector3 to = transform.position - transform.transform.position;
             return (Vector3.Angle(-transform.transform.forward, to) < 70f);
         }
         return false;
     }
 
     public void DummyNapeHit(DummyTitan titan)
     {
-        Vector3 vector3 = this.currentCamera.GetComponent<IN_GAME_MAIN_CAMERA>().main_object.GetComponent<Rigidbody>().velocity;
-        int num2 = (int)((vector3.magnitude * 10f) * this.scoreMulti);
-        num2 = Mathf.Max(10, num2);
+        Vector3 velocity = body.velocity;
+        int damage = (int) ((velocity.magnitude * 10f) * scoreMulti);
+        damage = Mathf.Max(10, damage);
 
-        titan.photonView.RPC("GetHit", PhotonTargets.All, new object[] { num2 });
-        GameObject.Find("MultiplayerManager").GetComponent<FengGameManagerMKII>().netShowDamage(num2);
+        titan.photonView.RPC(nameof(DummyTitan.GetHit), PhotonTargets.All, new object[] { damage });
+        manager.netShowDamage(damage);
     }
 
-    public void clearHits()
+    public void ClearHits()
     {
-        this.currentHitsII = new ArrayList();
-        this.currentHits = new ArrayList();
+        currentHitsII = new ArrayList();
```

Comment:
Rather use .Clear() than using creating a new ArrayList

Reply:
Though the difference might be minimal, I remember reading that `new ArrayList()` performed better vs `.Clear()`.